Regression: ReadableStream pipeThrough() assertion failed

Seems like pipeThrough crashes when given just one argument starting in Deno 1.7.0.

```typescript
const resp = await fetch("https://example.com");
resp.body.pipeThrough(new TransformStream());
```

```
error: Uncaught (in promise) AssertionError: Assertion failed.
      throw new AssertionError(msg);
            ^
    at assert (deno:op_crates/fetch/11_streams.js:25:13)
    at readableStreamPipeTo (deno:op_crates/fetch/11_streams.js:1223:5)
    at ReadableStream.pipeThrough (deno:op_crates/fetch/11_streams.js:3148:23)
    at file:///$deno$eval.ts:1:65
    at async file:///$deno$eval.ts:1:1
```

Workaround of specifying all the options explicitly:

```typescript
const pipeOpts: PipeOptions = {
  preventAbort: false,
  preventCancel: false,
  preventClose: false,
};

const resp = await fetch("https://example.com");
resp.body.pipeThrough(new TransformStream(), pipeOpts);
```

Copy-paste repro:

```
> docker run --rm hayd/alpine-deno:1.6.2 eval 'await fetch("https://example.com").then(x => console.log(x.body.pipeThrough(new TransformStream())))'
ReadableStream { locked: false }

> docker run --rm hayd/alpine-deno:1.6.3 eval 'await fetch("https://example.com").then(x => console.log(x.body.pipeThrough(new TransformStream())))'
ReadableStream { locked: false }

> docker run --rm hayd/alpine-deno:1.7.0 eval 'await fetch("https://example.com").then(x => console.log(x.body.pipeThrough(new TransformStream())))'
error: Uncaught (in promise) AssertionError: Assertion failed.
      throw new AssertionError(msg);
            ^
    at assert (deno:op_crates/fetch/11_streams.js:25:13)
    at readableStreamPipeTo (deno:op_crates/fetch/11_streams.js:1223:5)
    at ReadableStream.pipeThrough (deno:op_crates/fetch/11_streams.js:3148:23)
    at file:///$deno$eval.ts:1:65
    at async file:///$deno$eval.ts:1:1
```

Comes up in btoa wpt tests as-well.

> Internal Error: AssertionError: Assertion failed.
4864
    at assert (deno:op_crates/fetch/11_streams.js:25:13)
4865
    at writableStreamFinishInFlightCloseWithError (deno:op_crates/fetch/11_streams.js:2762:5)
4866
    at deno:op_crates/fetch/11_streams.js:2489:7

--- a/cli/tests/wpt.jsonc
+++ b/cli/tests/wpt.jsonc
@@ -8,7 +8,7 @@
     "piping/flow-control",
     // "piping/general",
     "piping/multiple-propagation",
-    // "piping/pipe-through",
+    "piping/pipe-through",
     "piping/then-interception",
     // "piping/throwing-options",
     // "piping/transform-streams",
@@ -134,8 +134,12 @@
     // TODO(lucacasonato): uses XMLHttpRequest unnecessarily. should be fixed upstream before enabling
     // "unsupported-encodings",
   ],
-  "dom": ["abort/event"],
-  "hr-time": ["monotonic-clock"],
+  "dom": [
+    "abort/event"
+  ],
+  "hr-time": [
+    "monotonic-clock"
+  ],
   "html": [
     "webappapis/microtask-queuing/queue-microtask-exceptions.any",
     "webappapis/microtask-queuing/queue-microtask.any",
@@ -183,7 +187,6 @@
     "webapi/status",
     "webapi/create_multiple_memory",
     "create_multiple_memory"
-
     //FAILING TESTS
     // "jsapi/constructor/instantiate-bad-imports",
     // "jsapi/constructor/instantiate",
@@ -225,7 +228,9 @@
     "console-namespace-object-class-string",
     "console-tests-historical"
   ],
-  "WebCryptoApi": ["getRandomValues"],
+  "WebCryptoApi": [
+    "getRandomValues"
+  ],
   "WebIDL": [
     "ecmascript-binding/es-exceptions/DOMException-constants",
     "ecmascript-binding/es-exceptions/DOMException-constructor-and-prototype",

--- a/op_crates/fetch/11_streams.js
+++ b/op_crates/fetch/11_streams.js
@@ -3136,11 +3136,25 @@
      * @returns {ReadableStream<T>}
      */
     pipeThrough(
-      { readable, writable },
+      transform,
       { preventClose, preventAbort, preventCancel, signal } = {},
     ) {
+      if (!isReadableStream(this)) {
+        throw new TypeError("this must be a ReadableStream");
+      }
+      const { readable } = transform;
+      if (!isReadableStream(readable)) {
+        throw new TypeError("readable must be a ReadableStream");
+      }
+      const { writable } = transform;
+      if (!isWritableStream(writable)) {
+        throw new TypeError("writable must be a WritableStream");
+      }
       if (isReadableStreamLocked(this)) {
         throw new TypeError("ReadableStream is already locked.");
+      }
+      if (signal !== undefined && !(signal instanceof AbortSignal)) {
+        throw new TypeError("signal must be an AbortSignal");
       }
       if (isWritableStreamLocked(writable)) {
         throw new TypeError("Target WritableStream is already locked.");
@@ -3148,9 +3162,9 @@
       const promise = readableStreamPipeTo(
         this,
         writable,
-        preventClose,
-        preventAbort,
-        preventCancel,
+        Boolean(preventClose),
+        Boolean(preventAbort),
+        Boolean(preventCancel),
         signal,
       );
       setPromiseIsHandledToTrue(promise);

